client/my-sites/woocommerce/woop/landing-page.tsx

```diff
@@ -10,72 +11,45 @@ import Image04 from 'calypso/assets/images/woocommerce/woop-cta-image04.jpeg';
 import CtaSection from 'calypso/components/cta-section';
 import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
 import MasonryWave from 'calypso/components/masonry-wave';
-import { addQueryArgs } from 'calypso/lib/url';
+import useWooCommerceOnPlansEligibility from 'calypso/signup/steps/woocommerce-install/hooks/use-woop-handling';
 
 interface Props {
 	startSetup: () => void;
-	isFeatureActive: boolean;
-	upgradingPlan: Record< string, unknown >;
-	siteSlug: string;
+	siteId: number;
 }
 
 const images = [ { src: Image01 }, { src: Image02 }, { src: Image03 }, { src: Image04 } ];
 
-const WoopLandingPage: React.FunctionComponent< Props > = ( {
-	startSetup,
-	siteSlug,
-	isFeatureActive,
-	upgradingPlan,
-} ) => {
+const WoopLandingPage: React.FunctionComponent< Props > = ( { startSetup, siteId } ) => {
 	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce', href: `/woocommerce-installation` } ];
 	const ctaRef = useRef( null );
 
-	function onCTAHandler() {
-		if ( ! isFeatureActive && upgradingPlan?.product_slug ) {
-			return ( window.location.href = addQueryArgs(
-				{ redirect_to: window.location.href },
-				`/checkout/${ siteSlug }/${ upgradingPlan.product_slug }`
-			) );
-		}
-
-		startSetup();
-	}
+	const { hasBlockers, wpcomDomain } = useWooCommerceOnPlansEligibility( siteId );
 
-	const upgradeButtonText = upgradingPlan.product_name
-		? sprintf(
-				/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
-				__( 'Upgrade to %s' ),
-				upgradingPlan.product_name
-		  )
-		: __( 'Upgrade' );
-
-	let headlineText;
+	function onCTAClickHandler() {
+		if ( ! config.isEnabled( 'woop' ) ) {
+			// Trigger the current flow.
+			startSetup();
+		}
 
-	if ( isFeatureActive ) {
-		headlineText = __( 'Build exactly the eCommerce website you want.' );
-	} else {
-		headlineText = upgradingPlan.product_name
-			? sprintf(
-					/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
-					__( 'Upgrade to the %s plan and set up your WooCommerce store.' ),
-					upgradingPlan.product_name
-			  )
-			: __( 'Upgrade to set up your WooCommerce store.' );
+		// Rolling on the new woocommerce-on-plans flow.
+		page( `/start/woocommerce-install/confirm?site=${ wpcomDomain }` );
```

Comment:
This will redirect to the correct 0 step (in this case confirm). Lets us move the default step around later without updating it here.

```suggestion
		page( `/start/woocommerce-install/?site=${ wpcomDomain }` );
```